### DISCUSSION TITLE
ModuleNotFoundError: No module named '<some internal module>'

### DISCUSSION THREAD

**Discussion by tigerinus** [pyinstaller/pyinstaller#8216] [Help] [answered]
I thought `pyinstaller` automatically includes internal modules within the same folder, no?

My project folder structure looks like

```shell
$ tree -L 1
.
├── __init__.py
├── build
├── chat_model
├── core
├── data
├── database
├── dist
├── main.py
├── middleware
├── model
├── pyinstaller.spec
├── requirements.txt
├── routes
├── utils
└── venv
```

However, when running the code after packed by `pyinstaller`, I got

```python
$ ./dist/chat/server 
Traceback (most recent call last):
  File "Chat-Backup/main.py", line 9, in <module>
ModuleNotFoundError: No module named 'middleware'
[784840] Failed to execute script 'main' due to unhandled exception!
```

**Comment by rokm** [comment, marked as answer]
Not if you have an `__init__.py` file next to the entry-point script. Because in that case, we assume that the directory with entry-point script is actually a package, and add one directory higher to the search path. If there was no `__init__.py` file there (*is* that directory actually a package, considering you have a `venv` in it?), then the directory itself would be added to the search path, and `middleware` would be collected.

**Reply by tigerinus** [reply]
removing `__init__.py` solved the issue. Thanks!
